src/rss_etl.py

```diff
@@ -0,0 +1,175 @@
+from supabase import create_client, Client
+import feedparser
+import os
+import re
+from enum import Enum
+from dataclasses import dataclass
+
+"""
+File for extracting data from the RSS feeds of the PA Senate and House of Representatives
+and loading it into Supabase.
+
+The database tables updated by this script are:
+    Bills
+    Revisions
+"""
+
+# from dotenv import load_dotenv
+# load_dotenv()  # for local secret management with .env file
+
+senate_rss_feed = "https://www.legis.state.pa.us/WU01/LI/RSS/SenateBills.xml"
+house_rss_feed = "https://www.legis.state.pa.us/WU01/LI/RSS/HouseBills.xml"
+
+# sb_api_url = os.getenv("SUPABASE_API_URL")  # local secret management with .env file
+# sb_api_key = os.getenv("SUPABASE_API_KEY")  # local secret management with .env file
+sb_api_url = os.environ["SUPABASE_API_URL"]  # github actions secret management
+sb_api_key = os.environ["SUPABASE_API_KEY"]  # github actions secret management
+
+
+class LegislativeBody(Enum):
+    SENATE = "Senate"
+    HOUSE = "House"
+
+
+REVISIONS_OUTPUT_DICT = {
+    "bill_internal_id": "",
+    "printer_no": "",
+    "full_text_link": "",
+    "publication_date": "",
+    "revision_guid": "",
+    "description": "",
+}
+
+BILL_OUTPUT_DICT = {
+    "bill_number": "",
+    "legislative_session": "",
+    "session_type": "",
+    "chamber": "",
+    "legislative_id": "",
+}
+
+
+class BillIdentifier:
+    """
+    Extracts unique information from a bill's guid and embeds them in discrete properties
+    """
+
+    @staticmethod
+    def extract_bill_guid(revision_guid):
+        return re.search("BillGuid=(.*)", revision_guid).group(1)
+
+    @staticmethod
+    def determine_chamber(chamber_letter):
+        if chamber_letter == "H":
+            return "House"
+        elif chamber_letter == "S":
+            return "Senate"
+        else:
+            raise Exception("Invalid chamber letter in guid")
+
+    def __init__(self, guid):
+        self.guid = guid
+        self.bill_guid = re.search(r"(\d{4}\d[H|S]B\d+)P\d+", guid).group(1)
+        re_result = re.search(r"(\d{4})(\d)(H|S)B(\d+)P(\d+)", guid)
+        self.legislative_session = re_result.group(1)
+        self.session_type = re_result.group(2)
+        self.chamber = self.determine_chamber(re_result.group(3))
+        self.bill_number = re_result.group(4)
+        self.printer_number = re_result.group(5)
+
+
+class OutputRecord:
+
+    def __init__(self, supa_con, table_name, output_dict):
+        self.supa_con = supa_con
+        self.table_name = table_name
+        self.output_dict = output_dict.copy()
+
+    def has_existing_supabase_records(self, matching_columns):
+        # TODO: This might suffice for now, but selecting every row in the table is not a good long-term solution
+        statement = self.supa_con.table(self.table_name).select("*", count="exact")
```

Comment:
talked to chat GPT 4 about this. Easy to change

Simply use:
```python
statement=self.sup_con.table(self.table_name).select("count(*)")
for column in matching_columns:
    statement = statement.eq(column, self.output_dict[column])
    response = statement.execute()
    if response.status_code = 200:
        count = response.data[0]['count']
    else:
        raise Exception(f"""
        Failed to query supabase:
        status_code: {response.status_code}
        error_message: {response.error_message}
        """)
```

Reply:
How is that code different than the code that is already there? It looks like it executes the query multiple times so you make several db calls instead of just one

Reply:
Agreed with John -- this result doesn't look too different, and querying for every column individually would just add overhead. 
In retrospect, since it's just executing a where clause, it should be fine.

I do like the error handling, however, and will include that in my next commit.

Reply:
Actually, after trying to work it out, that particular error handling doesn't work. Honestly, when I had the API break while testing it, I found the errors sufficiently descriptive, so I'll leave that be for now.

Reply:
Sorry i made a mistake with the indentation in my suggestion. The `.execute()` should only run after the loop is done
statement=self.sup_con.table(self.table_name).select("count(*)")
```python
for column in matching_columns:
    statement = statement.eq(column, self.output_dict[column])
    response = statement.execute()
if response.status_code == 200:
    count = response.data[0]['count']
else:
    raise Exception(f"""
    Failed to query supabase:
    status_code: {response.status_code}
    error_message: {response.error_message}
    """)
```

Reply:
That still didn't work. statement.execute() doesn't return something with a "status_code" attribute. I think you're thinking of http status code, but this functions differently.